pkg/netpol/diff/diff.go

```diff
@@ -109,21 +109,27 @@ func (da *DiffAnalyzer) computeDiffFromConnlistResults(
 ) (ConnectivityDiff, error) {
 	workloadsNames1, workloadsNames2 := getPeersNamesFromPeersList(workloads1), getPeersNamesFromPeersList(workloads2)
 
-	// get disjoint ip-blocks from both configs
-	ipPeers1, ipPeers2 := getIPblocksFromConnList(conns1), getIPblocksFromConnList(conns2)
-	disjointPeerIPMap, err := eval.DisjointPeerIPMap(ipPeers1, ipPeers2)
+	// get disjoint src and dst ip-blocks from both configs
+	srcIPPeers1, dstIPPeers1 := getIPblocksFromConnList(conns1)
+	srcIPPeers2, dstIPPeers2 := getIPblocksFromConnList(conns2)
+	srcDisjointPeerIPMap, err := eval.DisjointPeerIPMap(srcIPPeers1, srcIPPeers2)
+	if err != nil {
+		da.errors = append(da.errors, newHandlingIPpeersError(err))
+		return nil, err
+	}
+	dstDisjointPeerIPMap, err := eval.DisjointPeerIPMap(dstIPPeers1, dstIPPeers2)
```

Comment:
is there a test demonstrating this refined disjoint computation for the diff output?

Reply:
yes, for example all diff tests involving `tests\ipblockstest` dir.
these changes were necessary after separating the IP-Blocks into two lists (src and dst) in connlist;
without those changes,  diff tests with a diff on ipBlocks will fail with following error  :
`missing peer from input disjointPeerIPMap` ;